float64 new_speed
---
bool success